frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -468,6 +468,7 @@ export type ResourceLinkProps = {
   truncate?: boolean;
   nameSuffix?: React.ReactNode;
   children?: React.ReactNode;
+  metadata?: any;
```

Comment:
It feels redundant to pass name/namespace/groupVersionKind _AND_ the full resource `metadata`. I'd propose an API where you can pass _EITHER_

1. name/namespace/gvk, _OR_
2. the full object

This could be the same `ResourceLink` component or two different components.

We need to keep (1) for backwards compatibility and because sometimes we don't have the full object. (2) allows us to check the labels for managed-by to handle this use case generally (and is more convenient if you do have the full object). Maybe for (1) we can have an optional `managedBy` prop as well.

I think also by default we'd hide the managed-by icon/tooltip and callers would opt-in with a `showManagedBy` or similar prop.